arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -151,40 +152,27 @@ def run_timerlat(
                     )
 
                 for line in timeseries_output:
-                    line_list = line.split()
-
                     # The first object in the line is a string, and it is possible for
-                    # it to have spaces in it. We'll try to cast the second object,
-                    # which should be the CPU number surrounded with [], to an int. If
-                    # that fails, we discard that object from the line_list.
-                    for i, _ in enumerate(line_list):
-                        if i == 1:
-                            try:
-                                cpu_val = re.compile(r"\[([0-9][0-9][0-9])\]")
-                                cpu = int(cpu_val.match(line_list[i]).group(1))
-                                break
-                            except (ValueError, AttributeError):
-                                line_list.pop(i)
-                            except IndexError as error:
-                                print(
-                                    "Unable to determine CPU number; Skipping time "
-                                    f"series collection: {error}\n{line}"
-                                )
-                                timeseries_dict = {}
-                                break
+                    # it to have spaces in it, so we split first on the expected "["
+                    # deliniator for the CPU number field.
+                    # FIXME: I can't be 100% sure that a "[" wont' appear in the string
+                    # that is the first object of the line.
+                    stripped_line = line.split("[")
+                    line_list = stripped_line[1].split()
 
                     # Because the tracer format is dependent on the underlying OS and
                     # cannot be controlled by the container, check the tracer output
                     # format and break gracefully if we don't recognize it
                     try:
-                        uptimestamp = float(line_list[3][:-1])
-                        timestamp = str(
+                        cpu = int(line_list[0][:-1])
+                        uptimestamp = float(line_list[2][:-1])
+                        timestamp = (
                             datetime.fromtimestamp(uptimestamp + uptime_offset)
                             .astimezone()
                             .isoformat()
                         )
-                        context = str(line_list[6])
-                        latency = int(line_list[8])
+                        context = str(line_list[5])
+                        latency = int(line_list[7])
 
                     except (IndexError, ValueError) as error:
                         print(
```

Comment:
Since what you're trying to do is recognition and parsing, why not do that with a regex?  Admitted, the expression itself is not easy to read, but that's a single sunk cost, and the remainder of the code is very easy and expressive (and you don't need a `try` block).  (Note that the `re.compile()` call should be moved above the loop, perhaps to module-scope, so that it is executed only once.)
```python
                     # Match a line which starts with zero or more spaces, at
                     # least one character between a pair of angle-brackets, a
                     # hyphen, a sequence of digits, zero or more spaces, a
                     # sequence of three digits (which we capture) between a
                     # pair of square-brackets, a space, two characters, a space,
                     # a sequence of digits followed by a dot followed by six
                     # digits (which we capture), a colon, a space, an octathorp
                     # followed by a sequence of digits, a space, "context", one
                     # or more spaces, a sequence of non-spaces (which we
                     # capture), "timer_latency", a space, a sequence of digits
                     # (which we capture), a space, and "ns".  E.g.,
                     # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
                     # or
                     #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
                     data_line = re.compile(r" *<[^>]+>-[0-9]+ *\[([0-9]{3})\] .. ([0-9]+\.[0-9]{6}): #[0-9]+ context +([^ ]+) timer_latency ([0-9]+) ns")

                      [...]

                     matches = data_line.match(line)
                     if matches:
                         (cpu, uptimestamp, context, latency) = matches.groups()
                         cpu = int(cpu)
                         uptimestamp = float(uptimestamp)
                         # `context` is already a string, so no conversion is required.
                         latency = int(latency)
                    else:
                        print("Unable to determine CPU number; Skipping time "
                            f"series collection:\n\t{line}")
                        timeseries_dict = {}
                        break

                    timestamp = (
                            datetime.fromtimestamp(uptimestamp + uptime_offset)
                            .astimezone()
                            .isoformat()
                        )
```